website_sharing_whole_class_mosaic_plot_translation.Rmd

```diff
@@ -0,0 +1,507 @@
+
+#### 2.33 先按年龄划分，再按音乐划分，再按喜爱划分。
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music, df)
+```
+
+### 2.34 划分的方向
+
+请注意，在前面的例子中，分割的方向如下。
+
+#### 2.341 年龄--横向分割
+
+#### 2.342 音乐--垂直分割
+
+#### 2.343 最喜欢的--横向分割
+
+这是默认的方向模式：交替方向从水平方向开始。因此，我们得到如下同样的图：
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music, 
+       direction = c("h", "v", "h"), df)
+```
+
+可以根据需要改变方向。例如，要创建一个doubledecker图，除了最后一个变量，所有的分割都是垂直的。
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music,
+       direction = c("v", "v", "h"), df)
+```
+
+需要注意的是，方向向量是按照拆分的顺序（年龄、音乐、喜爱），而不是按照公式中变量出现的顺序。在公式中，最后一个要拆分的变量要在”~”前先列出来。
+
+
+### 2.4 填充颜色
+
+填充颜色的分类是根据最后划分的维度——即因变量（本例中最喜欢的冰淇淋口味）。(如果不能正常运行，请更新到vcd的最新版本。)
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music, 
+       highlighting_fill = c("grey90", "cornflowerblue"),
+       df)
+```
+
+### 2.5 标签
+
+想要看关于标签选项的官方文档，请参见Strucplot框架中关于标签的说明。
+
+#### 2.51 旋转标签
+
+rot_labels = 向量设置了图形四边的旋转度数--而不是变量分割顺序--这个顺序是：上、右、下、左。(与典型的基本图形顺序不同!)默认是 rot_labels = c(0, 90, 0, 90)。
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music, 
+       labeling = vcd::labeling_border(rot_labels = c(45, -45, 0, 0)),
+       df)
+```
+
+#### 2.52 简称标签
+
+标签按照分割的顺序进行缩写（写作direction=）。缩写算法在消除元音后（如果需要的话），会返回return指定的字符数。
+
+关于更多的格式化选项，请参见>?vcd::labeling_border。
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music, 
+       labeling = vcd::labeling_border(abbreviate_labs = c(3, 1, 6)), 
+       df)
+```
+
+### 2.6 单元间距
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music,
+       spacing = vcd::spacing_equal(sp = unit(0, "lines")),
+       df)
+```
+
+想看更多细节, 请看 >?vcd::spacings
+
+#### 2.61 使用vcd::doubledecker中的 Mosaic 
+
+```{r}
+data(Arthritis, package = "vcd")
+vcd::doubledecker(Improved ~ Treatment + Sex, data=Arthritis)
+```
+
+```{r}
+vcd::doubledecker(Music ~ Favorite + Age, 
+                  xtabs(Freq ~ Age + Music + Favorite, df))
+```
+
+#### 2.62 使用ggplot中的马赛克图
+
+要在ggplot2框架中创建马赛克图，请使用ggmosaic包中的geom_mosaic()。
+
+https://cran.r-project.org/web/packages/ggmosaic/vignettes/ggmosaic.html
+
+### 2.7 理论
+
+### 2.8 何时使用
+
+当您想查看多个分类变量之间的关系时，请注意以下几点
+
+### 2.9 考虑因素
+
+#### 2.91 标签
+
+在马赛克图中，当有很多维度时，标签的可读性可能会较差。这可以通过以下方法来缓解： - 缩写名称 - 旋转标签。
+
+#### 2.92 纵横比
+
+* 长度比面积更容易判断，所以尽量使用相同宽度或高度的矩形。
+
+* 更高更瘦的矩形判断效果更好（因为我们更擅长区分长度而不是面积）。
+
+#### 2.93 矩形之间的间隙
+
+* 无间隙=最高效率
+
+然而，间隙可以帮助提高可读性，所以请尝试不同的组合。
+
+* 可在划分处有间隙
+
+* 可以在层次结构中改变间隙大小。
+
+#### 2.94 颜色
+
+* 对细分小组中的比率有好处
+
+* 显示残余（residual）
+
+* 强调特定群体
+
+### 2.10 其他资源
+
+安东尼-尤文的《用R进行图形数据分析》第七章
+
+本文第三部分（见以下详细内容）
+
+## 3 Part 3
+### 3.1 ggmosaic的基础知识
+
+* 用于分类数据的可视化。
+
+* 可以生成条形图、堆叠条形图、马赛克图和双层图。
+
+* 图是分层构造的，所以变量的排序是非常重要的。
+
+* 集成在ggplot2中的geom里，允许分面(facetting)和分层(layering)。
+
+### 3.2 创建gmosaic
+
+ggmosaic主要是使用ggproto和productplots包创建的。
+
+ggproto能够让你在自己的包中扩展ggplot2。
+
+* 使用了productplots包中的数据处理方法。
+
+* 为了绘制geom，需要计算xmin、xmax、ymin和ymax。
+
+### 3.3 ggplot2的缺陷
+
+ggplot2不能处理总体数量在变化的变量群。
+
+* 目前的解决方案：写成x=product(x1，x2)来同时读入变量x1和x2。
+
+* product函数。
+
+* 是为了wrapper list而设计的函数。
+
+* 允许它通过代码检测。
+
+这些限制也会导致标签的问题，但这些都可以手动修复。
+
+### 3.4 geom_mosaic：设定规则。
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.